packages/graph/behaviors/prefer-async.ts

```diff
@@ -0,0 +1,39 @@
+import { TimelinePipe } from "@pnp/core";
+import { InjectHeaders } from "@pnp/queryable";
+import { GraphQueryable, _GraphQueryable, graphGet } from "../graphqueryable.js";
+import { RichLongRunningOperation } from "@microsoft/microsoft-graph-types";
+
+export function PreferAsync(pollIntervalMs: number = 25000, maxPolls: number = 4): TimelinePipe {
+    return (instance: _GraphQueryable) => {
+        instance.using(InjectHeaders({ "Prefer": "respond-async" }));
+
+        instance.on.parse(async function (url, response, result) {
+            if (response.status === 202) {
+                const opLocation = response.headers.get("Location");
+                const opId = opLocation.split("/").at(-1);
+
+                let succeeded = false;
+                const statusQuery = GraphQueryable(instance, `operations/${opId}`);
+                for (let i = 0; i < maxPolls; i++) {
+                    await new Promise(resolve => setTimeout(resolve, pollIntervalMs));
+
+                    const status = await graphGet<RichLongRunningOperation>(statusQuery);
```

Comment:
For GET you should be able to invoke the queryable itself `await statusQuery<>()`